File: InterviewPrep/medium/81_searchInRotatedArrayII.go
/**
 *      file    :  81_searchInRotatedArrayII.go
 *      author  :  techiedheeraj
 *      created :  2020 Dec 03 01:44:42
 *      lastMod :  Thu Dec  3 01:44:42 2020
**/

func search(nums []int, target int) bool {

  var hm = make(map[int]int)

  for i, j := range nums {
    hm[j] = i
  }

  _, ok := hm[target]
  if ok == true {
    return true
  }

  return false
}

File: InterviewPrep/medium/81_searchInRotatedArrayII.cpp
/**
 *      file    :  81_searchInRotatedArrayII.go
 *      author  :  techiedheeraj
 *      created :  2020 Nov 21 02:47:51
 *      lastMod :  Sat Nov 21 02:47:51 2020
**/

class Solution {
public:
    bool search(vector<int>& nums, int target) {

      unordered_set<int>hashSet;
      for(auto &i: nums)
        hashSet.insert(i);

      return hashSet.find(target) != hashSet.end();

    }
}
